### DISCUSSION TITLE
Create new Google Meet and share URL on Slack

### DISCUSSION THREAD

**Discussion by ValentinH** [johnlindquist/kit#663] [Share]
A simple script to create a new Google Meet and paste the generated URL on the current Slack chat.

[Open meet in Script Kit](https://scriptkit.com/api/new?name=meet&url=https://gist.githubusercontent.com/ValentinH/9d2b6c1525f1eed38e6af964bad22406/raw/006894bf2da7b5b84d3afdc8fc08553eec26c136/meet.ts")

```js
// Menu: Meet
// Description: Start new meet and paste on Slack
// Author: Valentin Hervieu
// Twitter: @ValentinHervieu

import "@johnlindquist/kit"

browse('https://meet.new')
let count = 1
let tab = ''

while (count < 20) {
  await wait(500)
  tab = await getActiveTab("Brave")
  if (!tab.startsWith('https://meet.google.com/new') && !tab.startsWith('https://meet.new')) break
  count += 1
}

tab = tab.split('?')[0]

await copy(tab)
await focusWindow('Slack', '')
await setSelectedText(tab)
```

**Comment by johnlindquist** [comment]
Oooh, very cool. I had to research to see that Google Meet doesn't have a "create meeting" API which seems weird... Very nice workaround for what must be an annoying problem. 🏆

Re: `sleep`

`await wait(500)` is a global that will do the _exact_ same thing as your `sleep`.

https://github.com/johnlindquist/kit/blob/main/src/target/app.ts#L814

**Reply by ValentinH** [reply]
Nice! I was pretty sure there was something existing but I couldn't find it 🙈
